### PR TITLE
PCHR-3602 Hides disabled relationship Types from non-root admins

### PR DESCRIPTION
**Pull request by backstageel**
## Overview
Hides disabled relationship types from non-root admins.

## Before
selection_002

## After
selection_001

### CODE REVIEW

**Comment by mickadoo** [comment]
@backstageel there are some commits from PCHR-3608 in here too, did you maybe do some rebasing? Please update the PR so only commits from PCHR-3602 are part of it

File: uk.co.compucorp.civicrm.hrcore/CRM/HRCore/Hook/PageRun/RelationshipTypesFilter.php [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  /**
+   * Checks if this is the right page
+   *
+   * @param CRM_Core_Page $page
+   *
+   * @return bool
+   */
+  public function shouldHandle($page) {
+    global $user;
+    return $page instanceof CRM_Admin_Page_RelationshipType && !in_array('administrator', $user->roles);
```
Comment (mickadoo) [inline uk.co.compucorp.civicrm.hrcore/CRM/HRCore/Hook/PageRun/RelationshipTypesFilter.php]: This line is pretty long, can you extract `in_array('administrator', $user->roles)` into a variable like `$isRootAdmin`

Also there should be an empty line before `return`

File: uk.co.compucorp.civicrm.hrcore/CRM/HRCore/Hook/PageRun/RelationshipTypesFilter.php [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+   */
+  public function handle($page) {
+    if (!$this->shouldHandle($page)) {
+      return;
+    }
+
+    $rows = $page->get_template_vars('rows');
+    // remove disabled relationship types
+    foreach ($rows as $index => $row) {
+      if (CRM_Utils_Array::value('is_active', $row) == '0'&& in_array(CRM_Utils_Array::value('name', $row),['Case Coordinator is','Employee of','Head of Household for','Household member of'])) {
```
Comment (mickadoo) [inline uk.co.compucorp.civicrm.hrcore/CRM/HRCore/Hook/PageRun/RelationshipTypesFilter.php]: Did you run `civilint` on this file? I see a missing space after `'0'`. Also this line is very long, please extract the variables to make it more readable like 

`['Case Coordinator is','Employee of','Head of Household for','Household member of']`  => `$targetRelTypes`
`CRM_Utils_Array::value('is_active', $row) == '1'` => `$isActive`
`CRM_Utils_Array::value('name', $row)` => `$relationshipName`
`in_array($relationshipName, $targetRelTypes)` => `$isTargetType`

So you'll end up with a more readable conditional like

`if (!$isActive && $isTargetType)`

The variable names are just suggestions, change them to whatever you like

File: uk.co.compucorp.civicrm.hrcore/CRM/HRCore/Hook/PageRun/RelationshipTypesFilter.php [hunk 3]
```diff
@@ -13,9 +13,18 @@ public function handle($page) {
...
+    $targetRelTypes = [
+      'Case Coordinator is',
+      'Employee of',
+      'Head of Household for',
+      'Household member of',
+    ];
     // remove disabled relationship types
     foreach ($rows as $index => $row) {
-      if (CRM_Utils_Array::value('is_active', $row) == '0'&& in_array(CRM_Utils_Array::value('name', $row),['Case Coordinator is','Employee of','Head of Household for','Household member of'])) {
+      $isActive = CRM_Utils_Array::value('is_active', $row) == '0';
```
Comment (mickadoo) [inline uk.co.compucorp.civicrm.hrcore/CRM/HRCore/Hook/PageRun/RelationshipTypesFilter.php]: This variable name doesn't make sense, when I made the suggestions you would also need to change some of the logic

`$isActive = CRM_Utils_Array::value('is_active', $row) == '0';`

If `is_active` is 0 then it's not active, that's why I suggested changing this to `== '1'` and inverting the logic later

Comment (backstageel) [reply uk.co.compucorp.civicrm.hrcore/CRM/HRCore/Hook/PageRun/RelationshipTypesFilter.php]: Ok, changed to '1' and changed the condition

**Pull request merged**